os/services/ip64/ip64-dns64.c

```diff
@@ -101,13 +98,17 @@ ip64_dns64_6to4(const uint8_t *ipv6data, int ipv6datalen,
   struct dns_hdr *hdr;
 
   hdr = (struct dns_hdr *)ipv4data;
-  PRINTF("ip64_dns64_6to4 id: %02x%02x\n", hdr->id[0], hdr->id[1]);
-  PRINTF("ip64_dns64_6to4 flags1: 0x%02x\n", hdr->flags1);
-  PRINTF("ip64_dns64_6to4 flags2: 0x%02x\n", hdr->flags2);
-  PRINTF("ip64_dns64_6to4 numquestions: 0x%02x\n", ((hdr->numquestions[0] << 8) + hdr->numquestions[1]));
-  PRINTF("ip64_dns64_6to4 numanswers: 0x%02x\n", ((hdr->numanswers[0] << 8) + hdr->numanswers[1]));
-  PRINTF("ip64_dns64_6to4 numauthrr: 0x%02x\n", ((hdr->numauthrr[0] << 8) + hdr->numauthrr[1]));
-  PRINTF("ip64_dns64_6to4 numextrarr: 0x%02x\n", ((hdr->numextrarr[0] << 8) + hdr->numextrarr[1]));
+  LOG_DBG("dns64_6to4 id: %02x%02x\n", hdr->id[0], hdr->id[1]);
```

Comment:
These fields are `uint8_t`, do they not trigger compiler warnings?

If they don't, try installing clang-14 in the docker container and `make CLANG=1 TARGET=...`.

Reply:
No I am not getting any warnings on gcc or clang. I added a common `LOG_LEVEL_IP64` define and set it to LOG_LEVEL_DBG, and ran
```
user@88ca61990995:~/contiki-ng/examples/ip64-router$ make CLANG=1 TARGET=zoul BOARD=orion
```

iiuc the `x` modifier expects unsigned integer, and simply promotes the uint8 - do you expect this to trigger a warning?

With that said, this is probably a good opportunity to switch to fixed-width modifiers.